Bringing the tools up2date and adding some new functions

PickleRickVE opened the pull request:
Summary is in changed README.md

added showing ex-raid-eligible gyms as maroon, thanks to un1matr1x

PickleRickVE committed:
Add files via upload

PickleRickVE committed:
Add files via upload

PickleRickVE committed:
Add files via upload

PickleRickVE committed:
Update README.md

included recent rework

PickleRickVE committed:
Add files via upload

PickleRickVE committed:
Update README.md

PickleRickVE committed:
show ex-raid-eligible gyms as maroon

credits @un1matr1x

PickleRickVE committed:
show names of stops & gyms on popup

abakedapplepie commented on rdm-tools.php:
```diff
@@ -1337,10 +1320,26 @@ function loadData() {
       if (result.spawnpoints != null) {
         result.spawnpoints.forEach(function(item) {
           spawnpoints.push(item);
+          if (item.despawn_sec != null){
```
Instead of duplicating the marker generation here with a switch for despawn_sec null, we should do something like if (item.despawn_sec != null) { var color = blue } else { var color = red }

Then pass the var to the marker builder. currently youre checking twice for each point, and with spawns being the most prevalant data i'd like to keep it as efficient as possible

abakedapplepie commented on README.md:
```diff
@@ -4,24 +4,29 @@ Tools for managing RealDeviceMap data
-* You don't have to have spawns/stops visible on map to perform reporting anymore.
-* Revised optimization function - same super simple algorithm, might be better performing though
-* Changes to data storage, functions no longer loop through map objects, coordinates for gyms/stops/spawns are stored separately. Should help with performance
+* spawnpoint color is different for verified despawn time
```
can you fix this section's grammar/punctuation to match the rest of the document?

abakedapplepie commented on README.md:
```diff
@@ -43,12 +48,13 @@ The map has a variety of control buttons for performing different functions:
 * ![Manual route placement](assets/map-place-circle.png?raw=true) Enables/disables manual placement of route points. Click on the map to drop a route point in the routing layer based on view radius setting.
 * ![Import Nests](assets/map-import-nests.png?raw=true) Pulls nest data from OSM and places polygons in the current map bounds covering all parks.
-* ![Import Polygon](assets/map-import-polygon.png?raw=true) Import CSV or GeoJSON polygon data. GeoJSON can contain multiple polygons, each one will be placed individually.
+* ![Import Polygon](assets/map-import-polygon.png?raw=true) Import CSV or GeoJSON polygon data. GeoJSON can contain multiple polygons, each one will be placed individually. Choose wether to import as editable polygon(s) or nest(s).
```
s/wether/whether/g ;)

PickleRickVE committed:
Update README.md

PickleRickVE committed:
Update rdm-tools.php

PickleRickVE committed:
changed view of despawn_sec

Popups of spawnpoints now show Despawn time: hh:mm:ss instead of seconds.

Un1matr1x commented on rdm-tools.php:
```diff
@@ -2189,24 +2172,22 @@ function getSpawnData($args) {
-  
+    }
+
     $sql_spawn = "SELECT pokemon_id, COUNT(pokemon_id) as count FROM rdmdb.pokemon WHERE " . $points_string . " AND first_seen_timestamp >= ? GROUP BY pokemon_id ORDER BY count DESC" . $limit;
```
this sql-qry needs to be fixed ...
instead of `rdmdb.pokemon` use `pokemon`
this was wrong all along ... kudoo to f.e. DrKendall

PickleRickVE committed:
changed rdmdb.pokemon to pokemon

thx to un1matr1x

PickleRickVE committed:
added spawnpoints to unknown poi functionality

PickleRickVE committed:
added functionality for three different s2cell level - thx @alex

PickleRickVE committed:
corrected count of visible points an readme

PickleRickVE committed:
corrected rendering and comma error, moved button 'Get all nest reports' from settings to 'GO', did some visual changes and added count of points in circles (thx @xxleevo!)

PickleRickVE committed:
added selectable tileset in settings

PickleRickVE committed:
fixed copy to clipboard of get output

PickleRickVE committed:
cosmetic change of showing despawn time

PickleRickVE committed:
added language selection for pokemon names in settings, added french names

PickleRickVE committed:
added multilanguage support with external files

PickleRickVE committed:
added function to get to home location

PickleRickVE committed:
Add files via upload

PickleRickVE committed:
Update README.md

PickleRickVE committed:
added optional optimization for shortest possible route

PickleRickVE committed:
updated README.md

PickleRickVE committed:
Update rdm-tools.php

PickleRickVE committed:
outsourcing the config

PickleRickVE committed:
added count of circles in output

PickleRickVE committed:
improved calculation of shortest route

PickleRickVE committed:
reworked settings of s2cells with max-level 19 (performance) an locked 2nd & 3rd level to 14/17

PickleRickVE committed:
changed osm query

PickleRickVE committed:
changed s2cell function to re-allow L20 cells and improved performance

PickleRickVE committed:
removed a bug

PickleRickVE committed:
added spawnpoints without despawn time as option to optimize route function

PickleRickVE committed:
added searchfield for cities, deactivated zoomcontrol

PickleRickVE committed:
added import of administrative boundaries via OSM

PickleRickVE committed:
reactivated zoom control

PickleRickVE committed:
added count of pois in s2l14 cells with label on map and option in settings

PickleRickVE committed:
Merge remote-tracking branch 'origin/master'

PickleRickVE committed:
Revert "Merge remote-tracking branch 'origin/master'"

This reverts commit 0e49ef741b8a2d13bfdc5bb482e012ecb2b45925, reversing
changes made to d469c2de18bae28c47fa18ed3d3ea0564129f85f.

PickleRickVE committed:
changed way of importing polygon instances to single polygons, not a big one anymore

PickleRickVE committed:
added import for and displaying of wayfarer submissions including 20m range and reorganized into planning and wayfarer mode

PickleRickVE committed:
added support for specific csv format for importing submissions from https://gitlab.com/AlfonsoML/wayfarer

PickleRickVE committed:
changed marker for submissions and added the photo to the popup

PickleRickVE committed:
changed color of touched s2l15 cells to black for differentiating if l14 cells are active

PickleRickVE committed:
improved some code

PickleRickVE committed:
added option in config for own tileset, made some grammatical optimizations and corrected default settings

PickleRickVE committed:
added polygon export in PoracleJS format & copy function

PickleRickVE committed:
added option for hiding spawnpoints last updated before a specific date

PickleRickVE committed:
added passive mode for imported circle instances

PickleRickVE committed:
corrected naming fault

PickleRickVE committed:
corrected naming fault

PickleRickVE committed:
improved performance - all pois in bound were loaded before, even if not displayed, now only activated pois

PickleRickVE committed:
Update README.md

PickleRickVE committed:
Update README.md

PickleRickVE committed:
optimized komma handling in import function and increased zoom for s2 cells as a consequence of earlier performance improvement

PickleRickVE committed:
added function to place own POIs and made their markers draggable with 20m-range bound onto

PickleRickVE committed:
fixed spawnreport

PickleRickVE committed:
fixed copy function for circle output

PickleRickVE committed:
added average points per circle in output

PickleRickVE committed:
fixed count of points in circles and cells. Previously only visible points were counted, now all active and included points are counted even if they're out of bounds.
Count for points in polygons is now divided in gyms, pokestops and spawnpoints.
S2cells are only generated if in bounds.

PickleRickVE committed:
Export of multiple polygons at once in json-, poracle geofences- & pmsf/rdm nestlist-format or simple list of coordinates with option for naming polygons/nests and saving as file

PickleRickVE committed:
built in a color picker for instances instead of a text-field

PickleRickVE committed:
added color picker for polygon import

PickleRickVE committed:
solved issue #5 - tapping key 6 no longer zooms out

PickleRickVE committed:
added autogeneration of the correct circle size for raid instances depending on the specific latitude with the option in settings to choose between iv-, raid- or own radius

PickleRickVE committed:
added tileserver with satellite view

PickleRickVE committed:
(re)added option for custom radius when importing instances

PickleRickVE committed:
added individual handling of instances and multi select on export

PickleRickVE committed:
fixed handling for generated circles

PickleRickVE committed:
fixed handling for optimized circles

PickleRickVE committed:
complete rework and some great new functions

PickleRickVE committed:
added french translation

PickleRickVE committed:
removed old files

PickleRickVE committed:
fixed single polygon export for administrative units

PickleRickVE committed:
added check for missed quests with selectable instances

PickleRickVE committed:
changed to right version for missed quests

PickleRickVE committed:
changed circle generation to bootstrap without s2cells and added an option in settings to turn them on again

PickleRickVE committed:
added last update to popup of gyms & stops

PickleRickVE committed:
added consideration of 'deleted'-column of pokestop table

PickleRickVE committed:
added optional use of password protection via .htaccess - initial user/password is admin/admin For more see wiki.

PickleRickVE committed:
Added radius and coordinates to circle popup, added free circle tool & changed predefined tooltips for translation

PickleRickVE committed:
Added circle size for 1gb devices in settings.

PickleRickVE committed:
fixed unknown points function and generally optimized loading of data

PickleRickVE committed:
fixed a bug that prevented output when deleting a circle from bootstrap instance

PickleRickVE committed:
fixed getting to the wrong world when completely zoomed out

PickleRickVE committed:
added some osm tags for nest identification

lenisko committed:
add path counter and lat,lon to circle popup

lenisko committed:
rename to Circle Counter - as it's not  counter

lenisko committed:
don't save counter in passive mode

123FLO321 committed:
config.env.php

123FLO321 committed:
use config.env.php as default config

123FLO321 committed:
Update config.env.php

PickleRickVE committed:
Merge pull request #10 from lenisko/circle_popup_addition

Add route counter, lat & lon to circle popup

PickleRickVE committed:
adjusted circle popup

PickleRickVE committed:
Merge pull request #12 from 123FLO321/docker

Support for config from Environment

PickleRickVE committed:
Delete config.example.php

PickleRickVE committed:
preserve osm id and name of manually imported nests or polygons

PickleRickVE committed:
support for pmsf's manualdb | read & write nests | write nestpokemon to db from nestreport or automated for all nests | several improvements and corrections

PickleRickVE committed:
added length for instance and area for polygons on the associated pop-ups

PickleRickVE committed:
adjustable options/query for importing nests from osm, function moved to nestoptions dialog

PickleRickVE committed:
fixed and added some parts of polygon import

PickleRickVE committed:
added automatic calculation of actual nest migration date

PickleRickVE committed:
fixed import of polygon instances

PickleRickVE committed:
added circle_smart_pokemon instance type

PickleRickVE committed:
Fix: added option in config to switch manualdb support on or off, standard is off to prevent malfunction for those without manualdb.

PickleRickVE committed:
added function to merge polygons and changed simple export of polygons

Deses committed:
Fixed several errors where empty lines or garbage input would crash the entire tool.

Deses committed:
Fixed errors in the polygon section

Deses committed:
Made the comparators more strict.
Made the modal hide itself after success just like the polygon modal.

Deses committed:
Added one more level to the Administrative Boundaries. Some regions are divided in level 7 regions. For example, Spain's "comarcas".

Deses committed:
Removed a bunch of duplicated code.

Deses committed:
Fix to Circle Radius Calculator

Deses committed:
More cleanup.

Deses committed:
Added an option (by default disabled) to show or hide S2 cells on big circles. By doing this, we DRASTICALLY reduce the amount of memory used and time taken when importing "gym", "1g" or similarly sized circles. I don't know why by default it showed them earlier, I don't see the use, but anyway if you enable the option in settings, you'll get the old funcionality back.

Deses committed:
Fixed a regression.

Deses committed:
Found another regression.

Deses committed:
bruh.

Deses committed:
Enabling the delete feture button could be useful. Way faster than having to click each layer.

Deses committed:
More small optimizations.

Deses committed:
Clear the bootstrap layer (initial circle array) when optimizing to remove a bunch of unnecesary stuff from the screen, freeing memory in the process. Also, it looks way cleaner this way.

Deses committed:
$("#modalLoading").modal('hide'); was in the wrong place, so it was never showing.
Also moved the clearLayer from the previous commit to the complete function.

PickleRickVE committed:
Merge pull request #17 from Deses/fix-import-empty-lines

Fix import empty lines

Fabio1988 committed:
update config.php usage

PickleRickVE committed:
Merge pull request #18 from Fabio1988/patch-2

update config.php usage

PickleRickVE committed:
added support for 9x9 cell scanning

PickleRickVE committed:
Add files via upload

ReuschelCGN committed:
Update index.php

ReuschelCGN committed:
Update index.php

PickleRickVE committed:
Merge pull request #19 from ReuschelCGN/master

lift fontawesome version & fix missing icon & change icons for better understanding

PickleRickVE committed:
removed old 1gb device support and added v2.0 support instead plus fixes for generating circle s2 cells

PickleRickVE committed:
added import for 9x9 instances via checkbox

PickleRickVE committed:
changed method for selection of import instance type to radio buttons

MC0R3 committed:
Docker Support updates & README changes

MC0R3 committed:
Update README.md

MC0R3 committed:
Update README.md

MC0R3 committed:
Update docker-compose.example.yml

Password support

MC0R3 committed:
Update Dockerfile

Password support

PickleRickVE committed:
added bootstrapping for 9x9

PickleRickVE committed:
changed spawnpoint query from updated to last_seen and added scrub as option for nest import.

PickleRickVE committed:
Merge pull request #21 from MC0R3/master

update for docker support

Fabio1988 committed:
create docker workflow

PickleRickVE committed:
Merge pull request #22 from Fabio1988/patch-3

create docker workflow

PickleRickVE committed:
fixed 9x9 bootstrap

Fabio1988 committed:
Update docker-compose.example.yml

PickleRickVE committed:
Merge pull request #23 from Fabio1988/patch-3

Update docker-compose.example.yml
